maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/booterclient/ForkStarter.java

```diff
@@ -162,7 +162,7 @@
 
         private final Thread inputStreamCloserHook;
 
-        public CloseableCloser( int jvmRun, Closeable... testProvidingInputStream )
+        CloseableCloser( int jvmRun, Closeable... testProvidingInputStream )
```

Comment:
why removing public constructor this is not backward compat???? if checkstyle complains about this so remove checkstyle usage as it's useless....

Reply:
This constructor wasn't used outside the package. Intellij also confirms it, so checkstyle reports it properly. Also keep in mind that tests passes. If it not being used, access can be safly lowered to package level.